xiilib/database_migration.py

```diff
@@ -67,15 +69,22 @@ def _set_status(self, status: DatabaseMigrationStatus) -> None:
         """
         self._container.push(self._status_file, source=status, make_dirs=True)
 
-    def run(
-        self, command: list[str], environment: dict[str, str], working_dir: pathlib.Path
+    def run(  # pylint: disable=too-many-arguments
```

Comment:
It's normal for an equivalent of `subprocess.run` to have this many arguments, as the real `subprocess.run` would have even more. This is simply a requirement of the function, this approach has been applied to `subprocess.run` as well as to `ops.Container.exec`.

There are some attempts in the industry that try to solve this issue, for example [golang's `exec.Cmd`](https://pkg.go.dev/os/exec#Cmd). But I don't this can be solved by `CharmState`, as this class needs to be utilized by various classes, such as `DjangoCharm`, `FlaskCharm`, and `SpringBootCharm` which means `CharmState` will be different for each charms.

Reply:
Indeed, that method is just a wrapper around `Container.exec` so that's fine. Probably adding a comment that explains that we disable `too-many-arguments` because the method is a wrapper around `Container.exec` that uses all these arguments